packages/api/models/schemas.js

```diff
@@ -592,6 +592,17 @@ module.exports.provider = {
       title: 'Port',
       type: 'number'
     },
+    httpConfiguration: {
```

Comment:
Generically I think we should be doing name-spaced properties on providers, but I'm willing to be argued out of that.   Putting in the PR to spark conversation.

Reply:
I would tend to agree for a new version of the client, but as-is it feels a little inconsistent. How does that apply to things that are shared across some but not all protocols? e.g. not all providers use `username` and `password`, nor necessarily in the same way, We also have things like certificateURI which is HTTP-specific that should probably live in this new field. For that particular value, moving it would be fine as I think the audience is very niche, if there at all. Anything else, we are looking at potentially painful migration.

Reply:
Not that I'm saying no, in fact I would prefer moving `certificateUri` in this PR, but did want to add to the conversation.

Reply:
` I think the audience is very niche, if there at all. Anything else, we are looking at potentially painful migration.`

Agreed in part.   *Migrations* are painful, although that's something we should probably address. 

In this PR I'm stuck between "start doing it the right way" and "make the mess worse when we *do* fix it" -- as there's a real user need I don't want to go too far down the rabbit hole either.   *Definitely* not before the team comes to a consensus on where we're headed with this.

Reply:
> I think the audience is very niche, if there at all.
I should clarify, this statement was referring specifically to the `certificateUri` field. I set that up mostly for our own testing and on the off-chance users need to use a self-signed cert or something, but I've not heard of any confirmed need for it.